### PR TITLE
feat(dicomImageLoader): make cornerstone to use new dicom image loader and handle scaling correctly

### PR DESCRIPTION
**Pull request by sedghi**
Blocked by https://github.com/Kitware/vtk-js/pull/2786 to get merged first 

--- 

Tried on various scenarios, multiple transfer syntaxes grayscale and color

- GPU
  - StackViewport -> with and without preferSizeOverAccuracy
  - Volume Viewport -> SAB or not

- CPU
  - StackViewport 

---

- StackViewport properly implements the preferSizeOverAccuracy
- Delegate data type decision to DICOMImageLoader based on the whether scaling should be needed or not 
- Fixed the libjpeg-turbo 8-bit
- Fixed the YBR JPEG images rendering
- Fixes the CPU rendering with Float results
- Fixed the CPU and window levelling
- Add new dicomImageLoader example loading various data 
- Changed examples in the dicom image loader -> examplesOld (just to keep them around for now)
- Make color images to use Uint8 instead of Uint8ClampedArray to avoid vtk that was converting it to float
- Calculate min and max inside the worker prior to deciding on the data type

### CODE REVIEW

**Comment by sedghi** [comment]
@Ouwen Could you please review this code as well?

File: packages/core/src/RenderingEngine/StackViewport.ts
Comment (wayfarer3130) [inline packages/core/src/RenderingEngine/StackViewport.ts:1436]: typedArray is the type of the thing, not what it is in terms of usage - please make this one pixelArray, as it is an array of pixels that happens to be of type TypedArray for general usage.

File: packages/dicomImageLoader/src/imageLoader/createImage.ts
Comment (wayfarer3130) [inline packages/dicomImageLoader/src/imageLoader/createImage.ts:237]: Please add a comment about vtk needing the Uint8Array here - it wouldn't be good to change that back unknowing.

File: packages/dicomImageLoader/src/imageLoader/createImage.ts
Comment (wayfarer3130) [inline packages/dicomImageLoader/src/imageLoader/createImage.ts:328]: Why is this one clamped while the other one isn't?

File: packages/dicomImageLoader/src/shared/getTypedArrayFromMinMax.ts
Comment (wayfarer3130) [inline packages/dicomImageLoader/src/shared/getTypedArrayFromMinMax.ts:2]: This function can return undefined - please have this start with:
let pixelArrayType = Float32Array;
Also, the function name implies you are getting a typed array, but what you are getting is the type of the pixel array, so please call it something like  getPixelArrayTypeFromMinMax   I got confused looking at the function expecting it to return a typed array instance.

File: packages/dicomImageLoader/src/shared/getTypedArrayFromMinMax.ts
Comment (wayfarer3130) [inline packages/dicomImageLoader/src/shared/getTypedArrayFromMinMax.ts:1]: Shouldn't the type of this by:
getPixelDataTypeFromMinMax(min: number, max: number): (() => TypedArray)) { ....
I think that will declare it as a funtion returning a constructor for TypedArray

**Review by wayfarer3130 — CHANGES_REQUESTED** [review]
There are a number of smallish changes that I'd like to see before approving/merging.

File: packages/dicomImageLoader/src/shared/getTypedArrayFromMinMax.ts
Comment (sedghi) [reply packages/dicomImageLoader/src/shared/getTypedArrayFromMinMax.ts:2]: There is an else statement which makes it Float32, typescript will yell if I do `let pixelArrayType = Float32Array;`

File: packages/dicomImageLoader/src/imageLoader/createImage.ts
Comment (Ouwen) [inline packages/dicomImageLoader/src/imageLoader/createImage.ts:202]: is this done after scaling?

File: packages/dicomImageLoader/src/shared/decodeImageFrame.ts
Comment (Ouwen) [inline packages/dicomImageLoader/src/shared/decodeImageFrame.ts:204]: `isColorImage`

File: packages/core/src/RenderingEngine/StackViewport.ts
Comment (wayfarer3130) [inline packages/core/src/RenderingEngine/StackViewport.ts:1600]: This line is meaningless - you never access pixelData after this point again.

File: packages/core/src/RenderingEngine/StackViewport.ts
Comment (wayfarer3130) [inline packages/core/src/RenderingEngine/StackViewport.ts:1576]: Please undo this line - it isn't needed as the assignment to pixelData is never used.

File: packages/core/src/RenderingEngine/StackViewport.ts
Comment (wayfarer3130) [inline packages/core/src/RenderingEngine/StackViewport.ts:1617]: Any way the image could be replaced rather than being modified in place?  Will this ever cause issues with other display environments needing rgba such as the thumbnails?

File: packages/core/src/RenderingEngine/StackViewport.ts
Comment (wayfarer3130) [inline packages/core/src/RenderingEngine/StackViewport.ts:1671]: This feels like a change that could be directly added to the Image class, rather than being in StackViewport, and potentially required for other Viewport types?

File: packages/core/src/RenderingEngine/StackViewport.ts
Comment (sedghi) [reply packages/core/src/RenderingEngine/StackViewport.ts:1617]: This is an edge case + optimization where image is rendered with rgba first, I guess I can make it not replace at all just uses the new pixelData, but every other call will do the same operation. Plus, I have a PR to make thumbnails use GPU soon

File: packages/core/src/RenderingEngine/StackViewport.ts
Comment (sedghi) [reply packages/core/src/RenderingEngine/StackViewport.ts:1671]: This is for CPU, image class doesn't care about GPU or CPU

File: packages/core/src/RenderingEngine/StackViewport.ts
Comment (wayfarer3130) [inline packages/core/src/RenderingEngine/StackViewport.ts:1733]: There is no need to create a secondary object to hold this return as toLowHighRange returns a new object every time.

File: packages/core/src/init.ts
Comment (wayfarer3130) [inline packages/core/src/init.ts:74]: Can you add a reason for commenting this out rather than removing it?

**Review by wayfarer3130 — APPROVED** [review]

**Comment by Ouwen** [comment]
🎆 🍾 🎉

**Pull request merged**
